client/core/libxc/binance.go

```diff
@@ -0,0 +1,1156 @@
+	// symbol field will be usdc.eth.
+	coin string
+	// network will be the same as coin for the base assets of
+	// a blockchain, but for tokens it will be the network
+	// that the token is hosted such as "ETH".
+	network          string
+	conversionFactor uint64
+}
+
+func bncSymbolData(symbol string) (*bncAssetConfig, error) {
+	coin := strings.ToUpper(symbol)
+	var ok bool
+	assetID, ok := dex.BipSymbolID(symbol)
+	if !ok {
+		return nil, fmt.Errorf("not id found for %q", symbol)
+	}
+	networkID := assetID
+	if token := asset.TokenInfo(assetID); token != nil {
+		networkID = token.ParentID
+		parts := strings.Split(symbol, ".")
+		coin = strings.ToUpper(parts[0])
+	}
+	ui, err := asset.UnitInfo(assetID)
+	if err != nil {
+		return nil, fmt.Errorf("no unit info found for %d", assetID)
+	}
+	return &bncAssetConfig{
+		assetID:          assetID,
+		symbol:           symbol,
+		coin:             coin,
+		network:          strings.ToUpper(dex.BipIDSymbol(networkID)),
+		conversionFactor: ui.Conventional.ConversionFactor,
+	}, nil
+}
+
+// bncBalance is the balance of an asset in conventional units. This must be
+// converted before returning.
+type bncBalance struct {
+	available float64
+	locked    float64
+}
+
+type binance struct {
+	wg          sync.WaitGroup
+	log         dex.Logger
+	url         string
+	wsURL       string
+	apiKey      string
+	secretKey   string
+	knownAssets map[uint32]bool
+	net         dex.Network
+	ctx         context.Context
+
+	markets atomic.Value // map[string]*symbol
+
+	balanceMtx sync.RWMutex
+	balances   map[string]*bncBalance
+
+	marketStreamMtx sync.RWMutex
+	marketStream    comms.WsConn
+
+	booksMtx sync.RWMutex
+	books    map[string]*bncBook
+
+	tradeUpdaterMtx    sync.RWMutex
+	tradeToUpdater     map[string]int
+	tradeUpdaters      map[int]chan *TradeUpdate
+	tradeUpdateCounter int
+
+	cexUpdatersMtx sync.RWMutex
+	cexUpdaters    []chan interface{}
+}
+
+var _ CEX = (*binance)(nil)
+
+func newBinance(apiKey, secretKey string, log dex.Logger, net dex.Network, binanceUS bool) CEX {
+	url, wsURL := httpURL, websocketURL
+	if binanceUS {
+		url, wsURL = usHttpURL, usWebsocketURL
+	}
+	if net == dex.Testnet || net == dex.Simnet {
+		url, wsURL = testnetHttpURL, testnetWebsocketURL
+	}
+
+	registeredAssets := asset.Assets()
+	knownAssets := make(map[uint32]bool, len(registeredAssets))
+	for _, a := range registeredAssets {
+		knownAssets[a.ID] = true
+	}
+
+	bnc := &binance{
+		log:            log,
+		url:            url,
+		wsURL:          wsURL,
+		apiKey:         apiKey,
+		secretKey:      secretKey,
+		knownAssets:    knownAssets,
+		balances:       make(map[string]*bncBalance),
+		books:          make(map[string]*bncBook),
+		net:            net,
+		tradeToUpdater: make(map[string]int),
+		tradeUpdaters:  make(map[int]chan *TradeUpdate),
+		cexUpdaters:    make([]chan interface{}, 0),
+	}
+
+	bnc.markets.Store(make(map[string]symbol))
+
+	return (CEX)(bnc)
```

Comment:
The mantra in Go that we flaunt often is "Accept Interfaces Return Structs".  
I think this should return `*binance` and the driver-like helper `NewCEX` that you've defined in interface.go can coerce it into an interface.  Although I'm not sure yet that it's the right place to force it into an interface.